lib/rules/jsx-max-props-per-line.js

```diff
@@ -29,8 +29,22 @@ module.exports = {
       type: 'object',
       properties: {
         maximum: {
-          type: 'integer',
-          minimum: 1
+          oneOf: [{
+            type: 'integer',
+            minimum: 1
+          }, {
+            type: 'object',
+            properties: {
+              single: {
+                type: 'integer',
+                minimum: 1
+              },
+              multi: {
+                type: 'integer',
+                minimum: 1
+              }
```

Comment:
are single/multi both required? what happens if i only pass one of them?

what about a value of zero? (as in, what if i want to force all props to be single-line, or all multi-line)

Reply:
No, other will default to Infinity.
So `maximum: {multi: 2}}` will apply maximum length to multiple line tags and ignore single line tags.

Zero value is a nice idea for another PR.  This rule originally didn't support such a case.